src/neuroconv/datainterfaces/behavior/audio/audiointerface.py

```diff
@@ -0,0 +1,203 @@
+            if not stub_test
+            else audio_data,
+        )
+
+        # Add metadata
+        acoustic_waveform_series_kwargs.update(**metadata)
+
+        # Create AcousticWaveformSeries with ndx-sound
+        acoustic_waveform_series = AcousticWaveformSeries(**acoustic_waveform_series_kwargs)
+
+        # Add audio recording to nwbfile as acquisition or stimuli
+        if write_as == "acquisition":
+            nwbfile.add_acquisition(acoustic_waveform_series)
+        elif write_as == "stimulus":
+            nwbfile.add_stimulus(acoustic_waveform_series)
+
+        return nwbfile
+
+    def run_conversion(
+        self,
+        nwbfile_path: OptionalFilePathType = None,
+        nwbfile: Optional[NWBFile] = None,
+        metadata: Optional[dict] = None,
+        stub_test: bool = False,
+        write_as: str = "stimulus",
+        starting_times: Optional[list] = None,
+        iterator_options: Optional[dict] = None,
+        compression_options: Optional[dict] = None,
+        overwrite: bool = False,
+        verbose: bool = True,
+    ):
+
+        file_paths = self.source_data["file_paths"]
+
+        assert write_as in ["stimulus", "acquisition"], "Audio can be written either as 'stimulus' or 'acquisition'."
+
+        if metadata is None:
+            metadata = dict()
+
+        audio_metadata = metadata.get("Behavior", dict()).get("Audio", None)
+        if audio_metadata is None:
+            audio_metadata = self.get_metadata()["Behavior"]["Audio"]
+
+        number_of_file_paths = len(file_paths)
+        assert len(audio_metadata) == number_of_file_paths, (
+            "Incomplete metadata "
+            f"(number of metadata in audio {len(audio_metadata)})"
+            f"is not equal to the number of file_paths {number_of_file_paths}"
+        )
+
+        audio_names = [audio["name"] for audio in audio_metadata]
+        audio_names_are_unique = len(set(audio_names)) == len(audio_names)
+        assert audio_names_are_unique, "Some of the names for AcousticWaveformSeries are not unique."
+
+        audio_metadata_unique, file_paths_unique = _check_duplicates(audio_metadata, file_paths)
+        unpacked_file_paths_unique = [file_path[0] for file_path in file_paths_unique]
+
+        if starting_times is not None:
+            assert isinstance(starting_times, list) and all(
+                [isinstance(x, float) for x in starting_times]
+            ), "Argument 'starting_times' must be a list of floats."
+
+            assert len(starting_times) == len(audio_metadata_unique), (
+                f"starting times list length {len(starting_times)} must be equal to number of unique "
+                f"AcousticWaveformSeries {len(audio_metadata_unique)} \n"
+                f"Audio metadata provided as input {audio_metadata} \n"
+                f"starting times = {starting_times} \n"
+                f"AcousticWaveformSeries after _check_duplicates {audio_metadata_unique}"
+            )
```

Comment:
Fair amount of metadata checking going on here - I'd prefer for this to be a subfunction (private or not) of the tools helper